### PR TITLE
Update dhcpd to work in shared network

### LINKED ISSUE #76: Patch DHCP server to work on a shared network
Currently DHCP server setup by helpernode denies request from unknown client (return DHCPNACK).

This makes nearly impossible to run multiple clusters on the same network where multiple DHCP servers will be running.

Proposing change: Remove the `deny unknown-clients` statement from dhcpd config to not NACK other cluster DHCPREQUEST on same network. Also delete the pool/range configuration where unknown-client won't be assigned with addresses from the allocation pool.

### PR DESCRIPTION
**Pull request by yussufsh**
Fixes #76

Signed-off-by: Yussuf Shaikh <yussuf.shaikh@ibm.com>

### CODE REVIEW

**Review by Prajyot-Parab — DISMISSED** [review]

File: playbooks/roles/ocp-config/README.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ Role Variables
 | chronyconfig.enabled    | no       | true           | Set to true to enable chrony configuration on the bastion node during installation. This also configure the bastion as a NTP server for the cluster. |
 | chronyconfig.content    | no       | ""             | List of time NTP servers and options pair (see chronyconfig examples). If empty, bastion will try sync with some default ntp server (internet) AND local HW clock (with higher stratum). |
 | chronyconfig.allow      | no       | ""             | List of network cidr (X.X.X.X/Y) allowed to sync with bastion configured as NTP server |
+| dhcp_shared_network     | no       |                | Flag to update DHCP server work on a shared network. (Neither ACK nor NACK unknown clients) |
```
Comment (bpradipt) [inline playbooks/roles/ocp-config/README.md:31]: Why to provide an option? Is there a problem to have it by default ?

File: playbooks/roles/ocp-config/tasks/dhcpd_update.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+# Update dhcpd config to not deny unknown clients and use only static entries
+- name: Find pool block
+  shell: grep 'pool {' /etc/dhcp/dhcpd.conf
+  ignore_errors: yes
+  register: dhcp_pool
+
+- name: Helm Configurations
```
Comment (bpradipt) [inline playbooks/roles/ocp-config/tasks/dhcpd_update.yaml]: Helm ??

Comment (yussufsh) [reply playbooks/roles/ocp-config/tasks/dhcpd_update.yaml]: Fixed.

File: playbooks/roles/ocp-config/README.md (hunk 1 above)
Comment (yussufsh) [reply playbooks/roles/ocp-config/README.md:31]: Just a preference.. if any issue we can easily disable this option.

File: playbooks/roles/ocp-config/tasks/dhcpd_update.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+---
+# Update dhcpd config to not deny unknown clients and use only static entries
+- name: Find pool block
+  shell: grep 'pool {' /etc/dhcp/dhcpd.conf
+  ignore_errors: yes
+  register: dhcp_pool
+
+- name: Update and restart dhcpd
```
Comment (bpradipt) [inline playbooks/roles/ocp-config/tasks/dhcpd_update.yaml]: "Update DHCP server config" ??

File: playbooks/roles/ocp-config/tasks/dhcpd_update.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
+# Update dhcpd config to not deny unknown clients and use only static entries
+- name: Find pool block
```
Comment (bpradipt) [inline playbooks/roles/ocp-config/tasks/dhcpd_update.yaml]: "Find DHCP Pool block" ?

File: playbooks/roles/ocp-config/tasks/main.yaml [hunk 5]
```diff
@@ -1,6 +1,10 @@
 ---
 # tasks file for ocp4 config
 
+- name: Update dhcpd to work on shared network
+  import_tasks: dhcpd_update.yaml
```
Comment (bpradipt) [inline playbooks/roles/ocp-config/tasks/main.yaml:5]: "Update DHCP server config for shared network"

**Review by bpradipt — APPROVED** [review]
/lgtm

**Review by Prajyot-Parab — APPROVED** [review]

**Pull request merged**